### PR TITLE
[Woo POS] Entry point eligibility conditions improvements

### LINKED ISSUE #11895: [Woo POS] Entry point eligibility conditions improvements
As turned out from investigation (p1719941118168819/1719932291.622679-slack-C070SJRA8DP) the current conditions eligibility conditions are not allowing to enter POS mode in case the device is a small tablet and in portrait mode. However, in landscape POS is enabled. We should take the device orientation into account, and show POS entry point in such case.

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #11895

### Description

The PR changes the screen size condition for entry to the POS mode. Only devices with:
* Smaller size >= 674
AND
* Bigger size >= 800

are eligible

pfoUAQ-Oq-p2

### Testing information

* Try different sizes of a screen
* Notice when POS is shown and when not

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/4923871/c1057fba-f1cc-46c7-a410-51c247fcc53e

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Please add tests for class <code>WooPosIsScreenSizeAllowed</code> (or add <code>unit-tests-exemption</code> label to ignore this).</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt [hunk 1]
```diff
@@ -40,11 +45,21 @@ sealed class WindowSizeClass(val maxWidth: Int) : Parcelable {
...
+     */
+    data object Large : WindowSizeClass(Int.MAX_VALUE)
 
     companion object {
         private const val COMPACT_SCREEN_MAX_WIDTH = 600
         private const val MEDIUM_SCREEN_MAX_WIDTH = 840
+        private const val EXPANDED_SCREEN_MAX_WIDTH = 1200
+    }
+
+    override fun compareTo(other: WindowSizeClass): Int {
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/extensions/ContextExt.kt:55]: That's something I had done before we changed the requirements, but I think it's nice to have that we can compare those

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/225">19.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/WooPosisFeatureFlagEnabled.kt [hunk 2]
```diff
@@ -3,7 +3,7 @@ package com.woocommerce.android.ui.woopos
 import com.woocommerce.android.util.FeatureFlag
 import javax.inject.Inject
 
-class IsWooPosFFEnabled @Inject constructor() {
+class WooPosisFeatureFlagEnabled @Inject constructor() {
```
Comment (samiuelson) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/WooPosisFeatureFlagEnabled.kt]:
```suggestion
class WooPosIsFeatureFlagEnabled @Inject constructor() {
```

**Review by samiuelson — APPROVED** [review]
LGTM! Works well, I only spotted a small typo.

**Pull request merged**